src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -89,15 +89,14 @@
     public App(AppConfig appConfig) {
         this.appConfig = appConfig;
 
-        ExecutorService collectionThreadPool =
-                buildExecutorService(appConfig.getThreadPoolSize());
-        collectionProcessor =
-                new TaskProcessor(collectionThreadPool, appConfig.getReporter());
-
-        ExecutorService recoveryThreadPool =
-                buildExecutorService(appConfig.getReconnectionThreadPoolSize());
+        ExecutorService collectionThreadPool = null;
+        ExecutorService recoveryThreadPool = null;
+        if (!appConfig.isEmbedded()) { // Creates executors in standalone mode only
+            collectionThreadPool = buildExecutorService(appConfig.getThreadPoolSize());
+            recoveryThreadPool = buildExecutorService(appConfig.getReconnectionThreadPoolSize());
+        }
         recoveryProcessor = new TaskProcessor(recoveryThreadPool, appConfig.getReporter());
-
+        collectionProcessor = new TaskProcessor(collectionThreadPool, appConfig.getReporter());
```

Comment:
instead of an embedded setting, perhaps it would be better to take an ExecutorService as an option?

What is the impact if it doesn't have these threadpools?

Reply:
I am not sure to follow your point here.
In embedded mode, collectionThreadPool & recoveryThreadPool variables are null, so TaskProcessor instances don't have any ExecutorService.

Reply:
Then what thread is jmxfetch even running on?

Reply:
The calling thread, i.e. the thread calling `App.run()` method.
In case of the tracer, this is the dd-jmx-collector thread created by the tracer.
see: https://github.com/DataDog/dd-trace-java/blob/master/dd-java-agent/agent-jmxfetch/src/main/java/datadog/trace/agent/jmxfetch/JMXFetch.java#L90